packages/html/src/utils/render.ts

```diff
@@ -8,14 +8,16 @@ import {CloudinaryVideo, CloudinaryImage} from "@cloudinary/url-gen";
  * @param pluginCloudinaryAsset {CloudinaryImage|CloudinaryVideo} The Cloudinary asset generated by base
  * @param plugins {plugins} array of plugins passed in by the user
  * @param pluginState {htmlPluginState} Holds cleanup callbacks and event subscriptions
- * @param analyticsOptions {AnalyticsOptions} analytics options for the url to be created
+ * @param analyticsOptions {BaseAnalyticsOptions} analytics options for the url to be created
  */
-export async function render(element: HTMLImageElement | HTMLVideoElement, pluginCloudinaryAsset: CloudinaryImage | CloudinaryVideo, plugins: Plugins, pluginState: HtmlPluginState, analyticsOptions?: AnalyticsOptions) {
+export async function render(element: HTMLImageElement | HTMLVideoElement, pluginCloudinaryAsset: CloudinaryImage | CloudinaryVideo, plugins: Plugins, pluginState: HtmlPluginState, analyticsOptions?: BaseAnalyticsOptions) {
     if(plugins === undefined) return;
     for(let i = 0; i < plugins.length; i++){
         const response = await plugins[i](element, pluginCloudinaryAsset, pluginState, analyticsOptions);
         if(response === 'canceled'){
             break;
+        } else {
+            return response
```

Comment:
The logic here seems strange to me.

The for-loop never makes it past the first iteration because it either breaks or returns.

Is this intentional behavior?